controllers/k8ssandra/k8ssandracluster_controller.go

```diff
@@ -86,6 +90,12 @@ func (r *K8ssandraClusterReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	patch := client.MergeFromWithOptions(kc.DeepCopy())
 	result, err := r.reconcile(ctx, kc, logger)
 	if kc.GetDeletionTimestamp() == nil {
+		if err != nil {
+			kc.Status.Error = err.Error()
+			r.Recorder.Event(kc, core.EventTypeWarning, "Reconcile Error", err.Error())
```

Comment:
It would be nice to see specific, concrete error types reported vs `Reconcile Error`. If I saw for the first time while using an operator I might think it is an unexpected error or bug in the controller. 

Note that we already have a `K8ssandraError` type. See [here](https://github.com/k8ssandra/k8ssandra-operator/blob/main/pkg/errors/errors.go#L20).

Reply:
I'm not sure I understand what you want. Would you have an example?
Seems like golang errors require a switch to check what type of error we're dealing with and we could get many different types here.